src/solver.rs

```diff
@@ -68,13 +68,15 @@
 use std::collections::BTreeSet as Set;
 use std::error::Error;
 use std::hash::Hash;
+use std::iter::FromIterator;
 
 use crate::error::PubGrubError;
 use crate::internal::core::State;
 use crate::internal::incompatibility::Incompatibility;
 use crate::internal::partial_solution::PartialSolution;
 use crate::package::Package;
 use crate::range::Range;
+use crate::solver::Dependencies::{Known, Unavailable};
```

Comment:
That's weird inside `solver.rs` in my opinion

Reply:
Yeah I made the same mistake in a sister PR, defining a type where I wanted to use it and not thinking more about general structure.

Where would you prefer it to be placed? Maybe a new `dependency` module, where we could move this and `DependencyProvider`?

Reply:
ah no I meant it's weird to use the crate you're in just to expose enum variants. But if you use them qualified, then you can remove this `use` line